### PR TITLE
Expanding E2E Test Cases

### PR DESCRIPTION
**Pull request by jaiveerk**
# Description
This PR is to add a full testing matrix for all combinations of cluster and dns zone type (private or public cluster with private or public dns zone).

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
E2E tests still run and test functionality as expected

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -24,16 +24,10 @@ jobs:
...
           ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_E2E_SUBSCRIPTION_ID }}
           ARM_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
-        run: "make dev-public"
-
-      - name: "Deploy operator"
-        run: "make push"
-
-      - name: "Run tests"
-        run: "make e2e"
+        run: "make all"
```
Comment (OliverMKing) [inline .github/workflows/e2e.yaml:33]: we should be using a matrix and testing all scenarios

Comment (OliverMKing) [reply .github/workflows/e2e.yaml:33]: that makes more sense than having a bash script that runs the matrix. we can run some things in parallel

**Review by OliverMKing — COMMENTED** [review]
Good stuff! Left one suggestion.

File: devenv/scripts/run_private_cluster.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+	set -e
+	echo "beginning private cluster tests..."
+	make dev CLUSTER_TYPE=private DNS_ZONE_TYPE=public
+	make push
+	make e2e
+
+  make dev CLUSTER_TYPE=private DNS_ZONE_TYPE=private
```
Comment (OliverMKing) [inline devenv/scripts/run_private_cluster.sh:7]: why is the indentation weird?

File: devenv/scripts/run_public_cluster.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+	set -e
+
+	echo "beginning public cluster tests..."
+	make clean
```
Comment (OliverMKing) [inline devenv/scripts/run_public_cluster.sh:4]: more weird indentation

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: devenv/scripts/run_private_cluster.sh (hunk 2 above)
Comment (jaiveerk) [reply devenv/scripts/run_private_cluster.sh]: looks fine on ide but got rid of indentation

File: devenv/scripts/run_public_cluster.sh (hunk 3 above)
Comment (jaiveerk) [reply devenv/scripts/run_public_cluster.sh]: looks fine on ide but got rid of indentation

**Review by OliverMKing — APPROVED** [review]

**Review by davidgamero — APPROVED** [review]
looks good!

**Pull request merged**